frontend/src/components/core/MainMenu/MainMenu.tsx

```diff
@@ -215,8 +220,108 @@ const MenuListItem = forwardRef<HTMLLIElement, MenuListItemProps>(
   }
 )
 
-function MainMenu(props: Props): ReactElement {
+const DevMenuListItem = forwardRef<HTMLLIElement, MenuListItemProps>(
+  (
+    {
+      item,
+      "aria-selected": ariaSelected,
+      onClick,
+      onMouseEnter,
+      $disabled,
+      $isHighlighted,
+    },
+    ref
+  ) => {
+    const { label, shortcut, hasDividerAbove } = item
+    const menuItemProps = {
+      isDisabled: $disabled,
+      isHighlighted: $isHighlighted,
+      isRecording: Boolean(item.stopRecordingIndicator),
+    }
+    const interactiveProps = $disabled
+      ? {}
+      : {
+          onClick,
+          onMouseEnter,
+        }
+
+    return (
+      <>
+        {hasDividerAbove && <StyledMenuDivider />}
+        <StyledDevMenuItem
+          ref={ref}
+          role="option"
+          aria-selected={ariaSelected}
+          aria-disabled={$disabled}
+          {...menuItemProps}
+          {...interactiveProps}
+        >
+          <StyledMenuItemLabel {...menuItemProps}>{label}</StyledMenuItemLabel>
+          {shortcut && (
+            <StyledMenuItemShortcut {...menuItemProps}>
+              {shortcut}
+            </StyledMenuItemShortcut>
+          )}
+        </StyledDevMenuItem>
+      </>
+    )
+  }
+)
+
+// eslint-disable-next-line
+const SubMenu = ({ menuOptions, closeMenu, isDevMenu }: any) => {
   const { colors }: Theme = useTheme()
+  if (!isDevMenu) {
+    return (
+      <StatefulMenu
+        items={menuOptions}
+        onItemSelect={({ item }) => {
+          item.onClick()
+          closeMenu()
+        }}
+        overrides={{
+          Option: MenuListItem,
+          List: {
+            props: {
+              "data-testid": "main-menu-list",
+            },
+            style: {
+              ":focus": {
+                outline: "none",
+              },
+              border: `1px solid ${colors.fadedText10}`,
+            },
+          },
+        }}
+      />
+    )
+  }
+  return (
+    <StatefulMenu
+      items={menuOptions}
+      onItemSelect={({ item }) => {
+        item.onClick()
+        closeMenu()
+      }}
+      overrides={{
+        Option: DevMenuListItem,
+        List: {
+          props: {
+            "data-testid": "main-menu-list",
+          },
+          style: {
+            ":focus": {
+              outline: "none",
+            },
+            border: `1px solid ${colors.fadedText10}`,
+          },
+        },
+      }}
+    />
+  )
+}
+
+function MainMenu(props: Props): ReactElement {
```

Comment:
Might be good to see if this entire function could be refactored since it's getting really hairy. On the other hand, doing that is out of scope for this work, so the right thing to do here may just be to file an issue to refactor this component later